### DISCUSSION TITLE
Import Vector Embeddings

### DISCUSSION THREAD

**Discussion by arkim822** [milvus-io/milvus#31049] [Q&A and General discussion] [answered]
Hello,  Milvus community. I have an array that contains vector embeddings and a list of values such as the text, dates, ids, and other metadata:

`
some_array = [
{'_additional': {'id': '0',
   'vector': [ a_long_list_of_floats] } ,
  'meta_data_value_1" : 'some_value',
  'meta_data_value_2": 'another_value'
},
... ,
{'_additional': {'id': '100,000',
   'vector': [ a_long_list_of_floats] } ,
  'meta_data_value_1" : 'some_value',
  'meta_data_value_2": 'another_value'
},
]
`

Is it possible to upload an object like this in a collection using the PyMilvus package?

I think I have found what I'm looking for here https://github.com/milvus-io/milvus/discussions/27243 but just want to make sure my logic is sound.

**Comment by xiaofan-luan** [comment]
<img width="481" alt="image" src="https://github.com/milvus-io/milvus/assets/83447078/534a50cc-e2e1-4856-a609-097d4d067ce6">

For current insert inerface, your insertion can be a pandas frame, or a dict.

**Comment by yhmo** [comment]
```
dimension = 512
fields=[
    FieldSchema(name="id", dtype=DataType.VARCHAR, is_primary=True, max_length=256),
    FieldSchema(name="vector", dtype = DataType.FLOAT_VECTOR, dim=dimension),
    FieldSchema(name="meta_data_value_1", dtype = DataType.VARCHAR, max_length=65535),
    FieldSchema(name="meta_data_value_2", dtype = DataType.VARCHAR, max_length=65535),
]

schema = CollectionSchema(fields=fields, enable_dynamic_field=False)
collection = Collection(name=collection_name, schema=schema)
print(f"Collection '{collection_name}' created")

# you can insert row by row
collection.insert({'id': '0', 'vector': [ a_long_list_of_floats], 'meta_data_value_1" : 'some_value', 'meta_data_value_2": 'another_value'})
collection.insert({'id': '100,000', 'vector': [ a_long_list_of_floats], 'meta_data_value_1" : 'some_value', 'meta_data_value_2": 'another_value'})

# you can insert a batch
collection.insert(
  [
    {'id': '0', 'vector': [ a_long_list_of_floats], 'meta_data_value_1" : 'some_value', 'meta_data_value_2": 'another_value'},
    {'id': '100,000', 'vector': [ a_long_list_of_floats], 'meta_data_value_1" : 'some_value', 'meta_data_value_2": 'another_value'
  ]
)
```

**Comment by arkim822** [comment]
Hello. I appreciate your replies. This is exactly what I wanted to confirm :) Thank you!

**Comment by arkim822** [comment]
Hello, again. Following up on this topic. I am using an embedding size of 1536 but I see here in this doc topic:
https://milvus.io/docs/create_collection.md
That there are only 3 options: [1, 32,768]
If that is the case then am I not able to search after uploading all of my embeddings?

**Reply by yiwangdr** [reply, marked as answer]
It's supposed to be [1, 32768], i.e. any number between 1 to 32768 should work. In your case, dim=1536 definitely should work.

**Reply by arkim822** [reply]
Okay, great! Thanks for the reply.
